assembl/static2/js/app/components/common/wordCloud/responsiveWordCloud.jsx

```diff
@@ -0,0 +1,79 @@
+// @flow
+import React, { Component } from 'react';
+
+// Component imports
+import ResizeAware from 'react-resize-aware';
+import WordCloud, { defaultBaseProps as defaultWordCloudBaseProps } from './wordCloud';
+
+// Type imports
+import type { BaseProps as WordCloudBaseProps } from './wordCloud';
+
+type Size = {
+  width: number,
+  height: number
+};
+
+class ResponsiveWordCloud extends Component<WordCloudBaseProps> {
+  static defaultProps = { ...defaultWordCloudBaseProps };
+
+  shouldComponentUpdate(nextProps: WordCloudBaseProps) {
+    const {
+      keywordsAngle,
+      keywordsColor,
+      keywordsColorActive,
+      keywords,
+      numberOfKeywordsToDisplay,
+      onMouseOutWord,
+      onMouseOverWord,
+      onWordClick
+    } = this.props;
+    if (
+      nextProps.keywordsAngle !== keywordsAngle ||
```

Comment:
You can have this in one line with ```return Object.keys(nextProps).some(key => nextProps[key] !== this.props[key])```

Reply:
But I'm not too sure what the point is here actually. Isn't a component already updating as soon as it receives a new props ?

Reply:
@MilanBarande in fact we pin point the props that should update the component, not every updated props should update the component

Reply:
(BTW I refactored the condition as you recommended)

Reply:
Alright makes sense